src/Features/CSharp/Portable/InlineParameterNameHints/InlineParameterNameHintsService.cs

```diff
@@ -0,0 +1,109 @@
+            var spans = new List<InlineParameterHint>();
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var nodes = root.DescendantNodes(textSpan);
+
+            foreach (var node in nodes)
+            {
+                if (node is ArgumentSyntax argument)
+                {
+                    if (argument.NameColon == null && IsExpressionWithNoName(argument.Expression))
+                    {
+                        var param = argument.DetermineParameter(semanticModel, cancellationToken: cancellationToken);
+                        if (param != null && param.Name != "")
+                        {
+                            spans.Add(new InlineParameterHint(param.Name, argument.Span.Start));
+                        }
+                    }
+                }
+                else if (node is AttributeArgumentSyntax attribute)
+                {
+                    if (attribute.NameEquals == null && attribute.NameColon == null && IsExpressionWithNoName(attribute.Expression))
+                    {
+                        var param = attribute.DetermineParameter(semanticModel, cancellationToken: cancellationToken);
+                        if (param != null && param.Name != "")
+                        {
+                            spans.Add(new InlineParameterHint(param.Name, attribute.SpanStart));
+                        }
+                    }
+                }
+            }
+            return spans;
+        }
+
+        /// <summary>
+        /// Determines if the argument is of a type that should have an adornment appended
+        /// </summary>
+        /// <param name="arg">The argument that is being looked at</param>
+        /// <returns>true when the adornment should be added</returns>
+        private static bool IsExpressionWithNoName(ExpressionSyntax arg)
+        {
+            if (arg is LiteralExpressionSyntax)
+            {
+                // We want to adorn literals no matter what
+                return true;
+            }
+            if (arg is ObjectCreationExpressionSyntax)
+            {
+                // We want to adorn object invocations that exist as arguments because they are not declared anywhere
+                // else in the file
+                // Example: testMethod(^new Object()); should show the adornment at the caret  
+                return true;
+            }
+            if (arg is CastExpressionSyntax cast)
+            {
+                // Recurse until we find a literal
+                // If so, then we should add the adornment
+                return IsExpressionWithNoName(cast.Expression);
+            }
+            if (arg is PrefixUnaryExpressionSyntax negation)
+            {
+                // Recurse until we find a literal
+                // If so, then we should add the adornment
+                return IsExpressionWithNoName(negation.Operand);
+            }
```

Comment:
note: i have a strong feelign this list will be incomplete.  i woudl recommend actually dogfooding your work here and seeing where you see these appear in roslyn and:

1. if you like where tehy appear.
2. if you feel lke it's missing from a place you would expect it to appear.